minject/model.py

```diff
@@ -34,6 +34,23 @@ class Resolver(abc.ABC):
     def resolve(self, key: "RegistryKey[T]") -> T:
         ...
 
+    @abc.abstractmethod
+    async def _aresolve(self, key: "RegistryKey[T]") -> T:
```

Comment:
Adding new abstract methods to an existing public base class is unfortunately a breaking change: it causes existing third-party implementations of that class to no longer implement the required methods.

To do this as a non-breaking change, perhaps you could provide default implementations that either return `Optional[T]` or raise `NotImplementedError`?

Reply:
Removing abstractmethod, raising `NotImplementedError`

Reply:
Chatted about this more with @mmchenry-duolingo. We will publish a patch bump shortly to remove everything outside of the current `minject.__init__.py` file from the public API.

We are treating it as a mistake that it was included.